### PR TITLE
add basic ui for submission page #52 N2

### PR DESCRIPTION
**Pull request by samyipsh**
sorry the the diffs for `answer.ts` is a bit hard to see because auto formatted on save.
 
only logic change was adding (`questionTitle`) field to api return of `getUserSubmissions`, also filtering out fields (`questionId`, `userId`).
- 'answers' was renamed to 'submission' for some api because terminology was quite confusing, i wan unsure if its the actual answer or the user answer, I referenced leetcode submissions page https://leetcode.com/submissions/#/1

TODO: 
- try modify api to return language as well

**Modified logic in answer.ts api**
``` getUserSubmissions: protectedProcedure.query(async ({ ctx }) => {
        const user = ctx.session.user;
        const submissions = await prismaPostgres.questionAttempt.findMany({
            where: {
                userId: user.id,
            },
        });

        const submissionsWithTitle = submissions.map(async (submission) => {
            const question = await prismaMongo.question.findUnique({
                where: { id: submission.questionId },
            });
            if (!question) {
                throw new TRPCError({
                    code: "BAD_REQUEST",
                    message: "Failed to find question",
                });
            }

            const { questionId, userId, ...filteredSubmission } = submission;
            return {
                ...filteredSubmission,
                questionTitle: question.title,
            };
        });

        return await Promise.all(submissionsWithTitle);
    }),
```

### CODE REVIEW

File: src/components/QuestionSubmissions.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import router from "next/router";
+import { useSession } from "next-auth/react";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+dayjs.extend(relativeTime);
+
+// TODO: modify to return submissions of a specific question
+// maybe of a specifc user (depending on where you want user to be able
+// to view all submission to a questions or just their submission)
+const QuestionSubmissions = ({ userId }: { userId: string }) => {
```
Comment (Gabau) [inline src/components/QuestionSubmissions.tsx:11]: Can use with auth wrapper for these?

File: src/pages/submissions/detail/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import Head from "next/head";
+import { useRouter } from "next/router";
+import { PageLayout } from "~/components/Layout";
+import { LoadingPage } from "~/components/Loading";
+import { api } from "~/utils/api";
+
+const SingleSubmissionDetailPage = (props: { submissionId: string }) => {
```
Comment (Gabau) [inline src/pages/submissions/detail/[id].tsx:7]: Same for this page

File: src/server/api/routers/answer.ts [hunk 3]
```diff
@@ -1,136 +1,188 @@
...
+
+  getAnswerBody: protectedProcedure
+    .input(
+      z.object({
+        answerId: z.string(),
+        // questionId: z.string(),
+      }),
+    )
+    .query(async ({ ctx, input }) => {
+      //   const user = ctx.session.user;
```
Comment (Gabau) [inline src/server/api/routers/answer.ts:160]: Remove these?

**Review by Gabau — APPROVED** [review]
Some comments

File: src/components/QuestionSubmissions.tsx (hunk 1 above)
Comment (samyipsh) [reply src/components/QuestionSubmissions.tsx:11]: hmm yea prolly should as discussed ill read up on that then replace everythign in one go yea

**Pull request merged**